lapack/lapack64/lapack64.go

```diff
@@ -466,6 +466,20 @@ func Lange(norm lapack.MatrixNorm, a blas64.General, work []float64) float64 {
 	return lapack64.Dlange(norm, a.Rows, a.Cols, a.Data, max(1, a.Stride), work)
 }
 
+// Langb returns the given norm of a square, n×n band matrix with kl sub-diagonals and
+// ku super-diagonals.
+//
+// When norm is lapack.MaxColumnSum, the length of work must be at least n.
+//
+// Dlangb is not part of the lapack.Float64 interface and so calls to Langb are always
+// executed by the Gonum implementation.
+func Langb(norm lapack.MatrixNorm, a blas64.Band, work []float64) float64 {
+	if a.Rows != a.Cols {
```

Comment:
This is unfortunate but there's no way around it. Lapack only computes norms of square band matrices. That means that this restriction would also propagate to `mat` and that's even more unfortunate. Since Lapacke doesn't provide this function, I think we're free to modify the signature. So that's one option.

On the other hand, Lapack only computes norms of square band matrices because it calls DLANGB only from two routines for solving square linear systems which I think is the principal utility of band matrices. So the other option would be to make a breaking change and allow only square band matrices in `mat`. Otherwise `mat` will never do anything useful with non-square band matrices anyway.

Reply:
I was too quick. The solve routines (DGBTRS, DGBSV, ...) take square matrices but the LU factorization (DGBTRF) of course works for general matrices. So there's only the first option.

Reply:
I've now pushed a commit that extends Dlangb to accept non-square band matrices.

Reply:
For completeness: I checked `mat.LU` and noticed that we limit the factorization only to square matrices. I remember we had a discussion on this topic in the past. Then the only remaining "native" operation on general, non-square banded matrices is MulVec supported by DGBMV. Still, it seems that it's more reasonable to leave banded matrices as general in `mat` and change Dlangb.

Reply:
This seems reasonable to me.